sky/shell/ui/flutter_font_selector.cc

```diff
@@ -32,9 +28,77 @@ using blink::FontData;
 using blink::FontDescription;
 using blink::FontFaceCreationParams;
 using blink::FontPlatformData;
+using blink::FontStyle;
+using blink::FontWeight;
 using blink::SimpleFontData;
 using mojo::asset_bundle::ZipAssetBundle;
 
+// Style attributes of a Flutter font asset.
+struct FlutterFontSelector::FlutterFontAttributes {
+  FlutterFontAttributes(const std::string& path);
+  ~FlutterFontAttributes();
+  std::string asset_path;
+  int weight;
+  FontStyle style;
+};
+
+// A Skia typeface along with a buffer holding the raw typeface asset data.
+struct FlutterFontSelector::TypefaceAsset {
+  TypefaceAsset();
+  ~TypefaceAsset();
+  RefPtr<SkTypeface> typeface;
+  std::vector<uint8_t> data;
+};
+
+namespace {
+const char kFontManifestAssetPath[] = "FontManifest.json";
+const int kFontWeightNormal = 400;
+
+const std::map<FontWeight, int> kFontWeightValue = {
+  {FontWeight::FontWeight100, 100},
+  {FontWeight::FontWeight200, 200},
+  {FontWeight::FontWeight300, 300},
+  {FontWeight::FontWeight400, 400},
+  {FontWeight::FontWeight500, 500},
+  {FontWeight::FontWeight600, 600},
+  {FontWeight::FontWeight700, 700},
+  {FontWeight::FontWeight800, 800},
+  {FontWeight::FontWeight900, 900},
+};
```

Comment:
This can just be an int[].  The FontWeight100 values are just integers from 0 to 8.